api/src/pcapi/core/offerers/models.py

```diff
@@ -622,6 +622,19 @@ def current_reimbursement_point(self) -> "Venue | None":
         link = self.current_reimbursement_point_link
         return link.reimbursementPoint if link else None
 
+    @property
+    def current_bank_account_link(self) -> "VenueBankAccountLink | None":
+        now = datetime.utcnow()
+
+        for link in self.bankAccountLinks:
+            lower = link.timespan.lower
+            upper = link.timespan.upper
+
+            if lower <= now and (not upper or now <= upper):
+                return link
+
+        return None
```

Comment:
Je me demande si on ne pourrais pas juste faire, vu qu’on ne fetch seulement les `bankAccountLinks` "actuels" (ou rien)

```suggestion
       if not self.bankAccountLinks:
            return None
       return self.bankAccountLinks[0]
```

Mais ça sous entend que le dev doit appeler cette property seulement en ayant correctement construit sa query, sous peine que ça renvoit un résultat erroné

Reply:
Ta suggestion me paraît trop dangereuse, car cette propriété doit fonctionner sans souci avec la backref sans présumer d'avoir personnalisé une requête. Si on veut y accéder comme tu le dis, alors on fait directement `self.bankAccountLinks[0] if self.bankAccountLinks else None` dans le code utilisant la requête.
Je laisse comme cela, de toute façon on n'aura dans notre cas zéro ou une seule itération de boucle, l'impact est donc anecdotique et permet de rester générique.